beacon-chain/rpc/eth/events/events.go

```diff
@@ -743,7 +743,7 @@ func (s *Server) fillEventData(ctx context.Context, ev payloadattribute.EventDat
 	if err != nil {
 		return ev, errors.Wrap(err, "could not get head state slot time")
 	}
-	ev.Attributer, err = s.computePayloadAttributes(ctx, st, hsr, ev.ProposerIndex, uint64(t.Unix()), randao)
```

Comment:
I don't think L713 should be called at all
When we call `ProcessSlotsUsingNextSlotCache`, instead of `hsr[:]`, we should pass in `ev.HeadRoot` here

Reply:
Done!

Reply:
Terence beat me to it - the issue wasn't that we were using the state root, it was that last block header doesn't have the state root value set (process_slots hasn't been called yet) so it is giving the wrong root. Thank you very much for the PR @ngotchac !